src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -0,0 +1,63 @@
+package reposense.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.Test;
+
+import reposense.parser.InvalidLocationException;
+import reposense.util.AssertUtil;
+
+public class RepoLocationTest {
+
+    @Test
+    public void repoLocation_parseGitUrl_success() throws Exception {
+        String gitUrl = "https://github.com/reposense/RepoSense.git";
+        RepoLocation repoLocation = new RepoLocation(gitUrl);
+        assertEquals(gitUrl, repoLocation.toString());
+        assertEquals("RepoSense", repoLocation.getRepoName());
+        assertEquals("reposense", repoLocation.getOrganization());
+        assertFalse(repoLocation.getParsedBranch().isPresent());
+
+        String gitUrlWithBranch = "https://github.com/reposense/testrepo-Alpha.git#release";
+        repoLocation = new RepoLocation(gitUrlWithBranch);
+        assertEquals("https://github.com/reposense/testrepo-Alpha.git", repoLocation.toString());
+        assertEquals("testrepo-Alpha", repoLocation.getRepoName());
+        assertEquals("reposense", repoLocation.getOrganization());
+        assertEquals("release", repoLocation.getParsedBranch().get());
+    }
+
+    @Test
+    public void repoLocation_parseInvalidGitUrl_throwsInvalidLocationException() {
+        // non GitHub url should rejected
+        AssertUtil.assertThrows(InvalidLocationException.class, () ->
```

Comment:
Should u assert the exception message as well? apart from the exception class

Reply:
Good point. Done